### ISSUE TITLE
Calling the configure max soc service switches the charging program to standard

### ISSUE THREAD

**Issue by josephdouce** [ReneNulschDE/mbapi2020#229]
### The problem

Calling the configure max soc service switches the charging program to standard.

Is this the desired behaviour? 

### What version of MBAPI2020 do use?

v0.13.0

### What was the last working version of MBAPI2020?

v0.13.0

### What type of installation are you running?

Home Assistant OS

### Diagnostics information

[config_entry-mbapi2020-8214be536ccc11f2e4386e0cdfb082d6.json](https://github.com/ReneNulschDE/mbapi2020/files/15387098/config_entry-mbapi2020-8214be536ccc11f2e4386e0cdfb082d6.json)

### Example YAML snippet

```yaml
alias: Charge Automation
description: ""
trigger:
  - platform: time_pattern
    minutes: "01"
  - platform: time_pattern
    minutes: "31"
condition:
  - condition: state
    entity_id: sensor.car_charging_program
    state: Home
action:
  - choose:
      - conditions:
          - condition: numeric_state
            entity_id: >-
              sensor.octopus_energy_electricity_acc_current_rate
            above: 0.15
        sequence:
          - service: mbapi2020.battery_max_soc_configure
            data:
              vin: !secret
              max_soc: "50"
      - conditions:
          - condition: numeric_state
            entity_id: >-
              sensor.octopus_energy_electricity_acc_current_rate
            below: 0.15
        sequence:
          - service: mbapi2020.battery_max_soc_configure
            data:
              vin: !secret
              max_soc: "100"
mode: single
```

### Additional information

I was hoping to use the "Home" profile to enable me to automate my dumb chargers, I can simply use standard however I would expect the update max soc to update the current charge program.

**Comment by ReneNulschDE** [comment]
No its not, I have not found a way to prevent this yet. I will keep it open and will dive deeper and check if I can use a different API to achieve the expected behavior.

**Comment by josephdouce** [comment]
Do you have a link to the API docs? I can't find a useful list of commands
anywhere 😩 the website is awful and gives 1 example and doesn't list any
of the commands, it's painful!

**Comment by ReneNulschDE** [comment]
There is no documentation as the API is not supported. There is just this [repo](https://github.com/mercedes-benz/MBSDK-CarKit-iOS). I will check it later this week.

**Comment by josephdouce** [comment]
Seems you can set maxsoc with the charge program configure, might be that
if you don't specify the charge program it defaults to 0.

**Comment by ReneNulschDE** [comment]
I have added the code to set the charge_program in the max_battery_configure service... Are you able to check the master branch code? Otherwise, it will be part of the beta release in the next a few days...

**Comment by josephdouce** [comment]
I checked the master branch, still switces to Standard when you call 

service: mbapi2020.battery_max_soc_configure
data:
  vin: VIN
  max_soc: "100"
  charge_program: "2"

**Comment by ReneNulschDE** [comment]
ok, let me check again...

**Comment by josephdouce** [comment]
![image](https://github.com/ReneNulschDE/mbapi2020/assets/1834595/ffefff2a-41fe-4084-98df-7bafbbe2a83f)

Looks like the profile isnt being passed to the function, 

If I change it to "charge_program: int" instead of "charge_program: int = 0" I get missing argument error.

**Comment by josephdouce** [comment]
See PR https://github.com/ReneNulschDE/mbapi2020/pull/248

**Comment by josephdouce** [comment]
Fixed with PR#248
